### PR TITLE
added usb support to proxmox module

### PR DESCRIPTION
**Pull request by rippleFCL**
##### SUMMARY

added usb support when managing proxmox kvm vms

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

proxmox_kvm

### CODE REVIEW

File: plugins/modules/proxmox_kvm.py [hunk 1]
```diff
@@ -517,6 +517,15 @@
         default: '2.0'
     type: dict
     version_added: 7.1.0
+  usb:
+    description:
+      - A hash/dictionary of usb devices for the VM. O(usb='{"key":"value", "key":"value"}').
```
Comment (felixfontein) [inline plugins/modules/proxmox_kvm.py:522]:
```suggestion
      - A hash/dictionary of USB devices for the VM. O(usb='{"key":"value", "key":"value"}').
```

File: plugins/modules/proxmox_kvm.py [hunk 2]
```diff
@@ -517,6 +517,15 @@
...
+      - Keys allowed are - C(usb[n]) where 0 ≤ n ≤ N.
+      - Values allowed are - C(host="value|spice",mapping="value",usb3="1|0").
+      - host is either C(spice) or the USB id/port.
+      - Option C(mapping) is the mapped USB device name.
+      - Option C(usb3) enables usb3 support.
```
Comment (felixfontein) [inline plugins/modules/proxmox_kvm.py:527]: Shouldn't this be
```suggestion
      - Option C(usb3) enables USB 3 support.
```
?

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I've added some first comments.

File: plugins/modules/proxmox_kvm.py (hunk 2 above)
Comment (rippleFCL) [reply plugins/modules/proxmox_kvm.py]: it should, ive changed both

File: plugins/modules/proxmox_kvm.py [hunk 3]
```diff
@@ -517,6 +517,15 @@
...
     version_added: 7.1.0
+  usb:
+    description:
+      - A hash/dictionary of USB devices for the VM. O(usb='{"key":"value", "key":"value"}').
+      - Keys allowed are - C(usb[n]) where 0 ≤ n ≤ N.
+      - Values allowed are - C(host="value|spice",mapping="value",usb3="1|0").
+      - host is either C(spice) or the USB id/port.
+      - Option C(mapping) is the mapped USB device name.
+      - Option C(usb3) enables USB 3 support.
+    type: dict
```
Comment (felixfontein) [inline plugins/modules/proxmox_kvm.py:528]:
```suggestion
    type: dict
    version_added: 8.6.0
```

**Review by felixfontein — COMMENTED** [review]
Please also add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment).

**Comment by felixfontein** [comment]
Please note that the commit you added adds a `version_added` directive, not a changelog fragment. A changelog fragment is a separate file in `changelogs/fragments/`. See the link above :)

File: changelogs/fragments/8199-added-usb-support-to-proxmox-module.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - "proxmox_kvm - adds``usb`` parameter for setting usb devices on proxmox kvm vms (https://github.com/ansible-collections/community.general/pull/8199)"
```
Comment (felixfontein) [inline changelogs/fragments/8199-added-usb-support-to-proxmox-module.yml:2]: According to https://en.wikipedia.org/wiki/Kernel-based_Virtual_Machine KVM is written upper-case. Also VM is an acronym, so it needs to be capitalized as well (https://docs.ansible.com/ansible/devel/dev_guide/style_guide/spelling_word_choice.html#acronyms). And then there's a period missing :)
```suggestion
  - "proxmox_kvm - adds``usb`` parameter for setting usb devices on proxmox KVM VMs (https://github.com/ansible-collections/community.general/pull/8199)."
```

File: changelogs/fragments/8199-added-usb-support-to-proxmox-module.yml [hunk 5]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - "proxmox_kvm - adds``usb`` parameter for setting usb devices on proxmox KVM VMs (https://github.com/ansible-collections/community.general/pull/8199)."
```
Comment (felixfontein) [inline changelogs/fragments/8199-added-usb-support-to-proxmox-module.yml:2]: Sorry, I missed that yesterday. I'll just push it and then merge once CI passed.
```suggestion
  - "proxmox_kvm - adds``usb`` parameter for setting USB devices on proxmox KVM VMs (https://github.com/ansible-collections/community.general/pull/8199)."
```

**Comment by felixfontein** [comment]
@rippleFCL thanks for your contribution!

**Pull request merged**